apple/swift-docc
# Allow writing doc links to subsections using special characters

Bug/issue #, if applicable: rdar://93458581

## Summary

This makes two changes related to doc links to subsections (doc links with fragments):
- Developers can now write links to subsections using characters that aren't allowed in URL fragments without needing to percent escape those characters in the link.
- Punctuation characters in doc links to subsections are now removed from the resolved topic reference fragment. This means that links to subsections written both with and without punctuation will resolve.

## Dependencies

None.

## Testing

In a DocC catalog with at least one article or documentation extension file:

- Add a few headings that contain punctuation characters
- Write links to those articles as `<doc:article-file-name#Heading-name-with-punctuation>`
The doc links to those subsections should resolve both with and without punctuation

- Add a few headings that contain emoji or other unicode characters
- Write links to those articles as `<doc:article-file-name#emoji-or-other-unicode-character>`
The doc links to those subsections should resolve without needing to percent encode the emoji or unicode characters.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
